src/HealthGPS/energy_balance_model.cpp

```diff
@@ -78,131 +82,169 @@ void EnergyBalanceModel::update_risk_factors(RuntimeContext &context) {
+    // Initial simulated person state.
     const double H_0 = person.get_risk_factor_value(H_key);
     const double BW_0 = person.get_risk_factor_value(BW_key);
     const double PAL_0 = person.get_risk_factor_value(PAL_key);
     const double F_0 = person.get_risk_factor_value(F_key);
     const double L_0 = person.get_risk_factor_value(L_key);
     const double ECF_0 = person.get_risk_factor_value(ECF_key);
+    const double G_0 = person.get_risk_factor_value(G_key);
     const double EI_0 = person.get_risk_factor_value(EI_key);
     const double CI_0 = person.get_risk_factor_value(CI_key);
-    const double G_0 = 0.5;
 
-    // TODO: Update height.
+    // TODO: Compute height.
     double H = H_0;
 
-    // TODO: Update physical activity level.
+    // TODO: Compute physical activity level.
     double PAL = PAL_0;
 
-    // Compute nutrient intakes from food intakes.
-    std::unordered_map<core::Identifier, double> nutrient_intakes;
-    for (const auto &coefficient : energy_equation_) {
-        nutrient_intakes[coefficient.first] = 0.0;
-    }
-    for (const auto &equation : nutrient_equations_) {
-        double food_intake = person.get_risk_factor_value(equation.first);
-
-        for (const auto &coefficient : equation.second) {
-            double delta_nutrient = food_intake * coefficient.second;
-            nutrient_intakes.at(coefficient.first) += delta_nutrient;
-        }
-    }
-
-    // Update energy intake and carbohydrate intake.
-    double EI = 0.0;
+    // Compute energy intake and carbohydrate intake.
+    auto nutrient_intakes = compute_nutrients(person);
+    double EI = compute_EI(nutrient_intakes);
     double CI = nutrient_intakes.at(CI_key);
-    for (const auto &coefficient : energy_equation_) {
-        double delta_energy = nutrient_intakes.at(coefficient.first) * coefficient.second;
-        EI += delta_energy;
-    }
 
-    // Update glycogen and water.
-    double k_G = CI_0 / (G_0 * G_0);
-    double G = sqrt(CI / k_G);
+    // Compute glycogen and water.
+    double G = compute_G(CI, CI_0, G_0);
     double W = 2.7 * G;
 
-    // Update extracellular fluid.
-    double Na_b = 4000.0;
-    double Na_f = Na_b * EI / EI_0;
-    double Delta_Na_diet = Na_f - Na_b;
-    double ECF = ECF_0 + (Delta_Na_diet - xi_CI * (1.0 - CI / CI_0)) / xi_Na;
+    // Compute extracellular fluid.
+    double ECF = compute_ECF(EI, EI_0, CI, CI_0, ECF_0);
 
-    // Update resting metabolic rate (Mifflin-St Jeor).
-    double RMR = 9.99 * BW_0 + 6.25 * H * 100.0 - 4.92 * person.age;
-    RMR += person.gender == core::Gender::male ? 5.0 : -161.0;
-    RMR *= 4.184; // kcal to kJ
+    // Compute resting metabolic rate (Mifflin-St Jeor).
+    double RMR = compute_RMR(BW_0, H, person.age, person.gender);
 
     double delta_0 = ((1.0 - beta_TEF) * PAL - 1.0) * RMR / BW_0;
 
-    // Update thermic effect of food and adaptive thermogenesis.
+    // Compute thermic effect of food and adaptive thermogenesis.
```

Comment:
I think this can still be split up a bit further. I think this function should basically not do any computation directly and should just call a series of `compute_*` functions then bundle the results together.